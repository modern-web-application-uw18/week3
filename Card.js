import React, { Component } from 'react';
// import AddItem from './AddItem.js';
import PropTypes from 'prop-types';

class Card extends Component {
  constructor(props){
    super(props);
    this.state = {
      items: [],
      cost: 0

  };

}

  addToCart = () => {
    // this.setState({ cost: this.state.cost + this.props.content.payment.cost });
    this.setState((prevState, props) => {
      const totalItems = prevState.items;
      totalItems.push(this.props.content);

      // console.log(this.props.content.payment.cost);
      // console.log(this.state.cost);


      return {
        items: totalItems,
        cost: this.state.cost + this.props.content.payment.cost
      };
    });
  }

  render() {
    // console.log(this.props.content);
    // console.log(this.props.content.image);
    console.log(this.state);
    // console.log(this.props);
    return (
      <div key={this.props.content.title} className='rental'>
        <img src={this.props.content.image} alt='stockImage'/>
        <div>
          <h5>{this.props.content.title}</h5>
          <p>Nightly rate: ${this.props.content.payment.cost}</p>
          {/*<AddItem onAddItemClick={this.addItem} />*/}
          <button onClick={this.addToCart}>Add to cart</button>
        </div>
      </div>
    );
  }
}

Card.PropTypes = {
  content: PropTypes.object.isRequired,
  payment: PropTypes.object.isRequired,
  title: Proptypes.string.isRequired,
  location: PropTypes.string
}

export default Card;
