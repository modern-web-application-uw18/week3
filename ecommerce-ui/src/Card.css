.Card-img {
	width: 430px;
}

.Card-title {
	font-size: 19px;
	font-weight: bold;
}

.flex-container {
  display: flex;
  flex-wrap: wrap;
  border-style: solid;
  border-width: 1px;
	border-color: lightgray;
	margin: 10px;
}

.flex-container > div {
  width: 500px;
	height: 450px;
  font-size: 15px;
}

.grid-card-top {
  margin: auto;
}

.grid-card-bottom {
	position: relative;
	margin: 9px;
}
