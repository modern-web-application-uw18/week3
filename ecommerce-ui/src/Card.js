import React, { Component } from 'react';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';
import './CardStyle.css';


class Card extends Component{  
     
    Superhost(props){
        const isSuperhost = this.props.card.host.isSuperhost;
         if(isSuperhost){
             return <p className='Superhost'>*Superhost</p>;
         } 
    }
    
    render(){
     
        return(
      <div className='Card-container'>   
           <div className='Card-img'>
                <img className = 'theImage' 
                        alt='staging'  
                        src={this.props.card.image}/>       
            </div>
            
            <div className='House-type-city-title'>
                <div className='Card-housetype'>
                    {this.props.card.houseType}
                        * 
                    {this.props.card.location.city}
                </div>
                    <div className='Card-title'>
                    {this.props.card.title}
                    </div> 
            </div>        
            <div className='Card-payment'>${this.props.card.payment.cost} per night * {this.props.card.payment.description} 
            </div> {/*end card payment  */}
            <div className='Card-rating'>
                 <StarRatings className='Stars' 
                    rating={this.props.card.rating.stars} 
                    starRatedColor='blue' 
                    numberOfStars={5}
                    starDimension='15px'
                    starSpacing="0.5px" 
                 /> 
                 
               <div className='Number-reviews'> 
               <div className='reviews'>{this.props.card.rating.reviews}</div>
               {this.Superhost(this.props.card.host.isSuperhost)}
               </div> 
            </div> {/*end card rating */} 
       </div>
        );
    }

}


Card.propTypes = {
    card: PropTypes.shape({  
        title: PropTypes.string,
        houseType: PropTypes.string,
        image: PropTypes.string,
        location: PropTypes.shape({
            city: PropTypes.string,
            country:PropTypes.string
        }),
        payment: PropTypes.shape({
            cost: PropTypes.number,
            description: PropTypes.string
        }),
        host: PropTypes.shape({
            name: PropTypes.string,
            isSuperhost: PropTypes.bool
        }),
        rating: PropTypes.shape({
            stars: PropTypes.number,
            reviews: PropTypes.number
        })
    })
};


export default Card;