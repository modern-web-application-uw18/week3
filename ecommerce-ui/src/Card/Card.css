.card {
  width: 30%;
  font-family: 'Roboto', sans-serif;
  border: 15px solid rgb(255, 255, 255);
  background: rgb(255, 255, 255);
}

.coverImage {
  height: 300px;
  position: relative;
}

.textOverlay {
  font-family: 'Play', sans-serif;
  background: rgba(0, 0, 0, .5);
  padding: 5px 30px;
  color: white;
  position: absolute;
  bottom: 50px;
  font-size: 25px;
}
