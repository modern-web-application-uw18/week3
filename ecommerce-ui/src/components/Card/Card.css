.Card {
    height: 200px;
    width: 500px;
    min-width: 40%;
    display: inline-flex;
    margin: 5px;
    background-position: 50% 50% !important;
    background-size: cover!important;
    text-align: left !important;
    padding: 5px;
    position: relative;
    flex-wrap: wrap;
    border: solid 1px #cecece;

}


.Title {
    background-color: #ffffff;
    color: black;
    width: 50%;
    height: 50%;
    padding: 5px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 25px;
    display: inline-flex;    
    opacity: 0.90;
}


.Rating {
    background-color: #be2c00;
    color: rgb(231, 231, 231);
    width: 50%;
    height: 18%;
    padding: 5px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 15px;
    display: inline-flex;    
    opacity: 0.90;
}


.UpperRight {
    background-color: #ffffff; 
    text-align: right;
    width: 48%;
    height: 50%;
    padding: 5px;
    font-family: 'Roboto', sans-serif;
    font-weight: 700;
    /* font-style: italic; */
    font-size: 45px;
    display: inline-flex; 
    margin-left: 5px;   
    opacity: 0.750;
}

.AlignRight {
    text-align: center; 
    width: 100%;
    color: #007e15;
    opacity: 1;

}

.LowerRight {
    background-color: #1b1a1a;
    color: rgb(231, 231, 231);
    width: 48%;
    height: 50%;
    padding: 5px;
    font-family: 'Roboto', sans-serif;
    font-weight: 300;
    font-style: italic;
    font-size: 15px;
    display: inline-flex; 
    margin-left: 5px;   
    opacity: 0.90;
}

.nopadding {
    padding-left: 0 !important;
    padding-right: 0 !important;
 }


@media (max-width: 650px) {
    .Card {
        width: 100%;
        margin: 0px;
    }

    .Title .Rating .UpperRight .LowerLeft {
        font-size: 20px;
    }


}
 
