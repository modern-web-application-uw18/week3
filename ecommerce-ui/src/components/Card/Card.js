import React from 'react';
import classes from './Card.css';
import MdAddShoppingCart from 'react-icons/lib/md/add-shopping-cart';
import Star from '../Star/Star';




const layout = (props) => (


  
    <div className={classes.Card} style={{ backgroundImage: `url(${props.item.image})` }}>

        <div className={classes.Title}>
            {props.item.title}
        </div>

        <div className={classes.UpperRight}>

            <div className={classes.AlignRight}>
                $ {props.item.payment.cost}
            </div>

        </div>

        <div className={classes.Rating}>
            <div style={{ width: '200px' }}>
                {props.item.houseType}
            </div>
            &nbsp;&nbsp;
            <Star starCount={props.item.rating.stars} />
            <span style={{ fontSize: '13px' }}>
                ({props.item.rating.reviews})
             </span>
        </div>


        <div className={classes.LowerRight}>


             <div>
                {props.item.location.city}, {props.item.location.city}
            </div>

            <div style={{ textAlign: 'left'}}>
                <div className={classes.Name}>
                  {props.item.host.name}
                </div>
            </div>

            <div style={{textAlign: 'right' }}>
                    <button type="button" className="btn btn-default" style={{width:'45px',height:'35px',backgroundColor:'#5b5b5b',border:'0px'}}><MdAddShoppingCart size={20} style={{color:'#ffffff'}} /></button>
            </div> 


        </div>

    </div>







    //         <div className={classes.Card} style={{ backgroundImage: `url(${props.item.image})` }}>

    //         <div className={classes.Title}>
    //         {props.item.title}
    //     </div> 

    // <div className='container' style={{paddingLeft: 0, paddingRight: 0}}>

    // <div className='container' className={classes.nopadding}>
    //         <div className='row'>
    //             <div className="col">
    //                 <div className={classes.Title}>
    //                     {props.item.title}
    //                 </div>
    //             </div>

    //             <div className="col">
    //             <div className={classes.UpperRight}>
    //                     {props.item.houseType}
    //                 </div>
    //              </div>
    //         </div>

    //         <div className='row'>
    //             <div className="col">

    //                 <div className={classes.Rating}>
    //                     {props.item.houseType}
    //                 </div>


    //             </div>

    //             <div className="col">
    //             <div className={classes.LowerLeft}>
    //                     {props.item.houseType}
    //                 </div>
    //              </div>
    //         </div>
    //     </div>
    //     </div>
    //     </div>

)


export default layout;
