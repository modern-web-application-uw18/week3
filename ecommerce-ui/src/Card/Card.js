import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './Card.css';

import HouseInfo from '../SectionBrowse/HouseInfo/HouseInfo.js';
import ArticleInfo from '../SectionArticles/ArticleInfo/ArticleInfo.js';
import SearchInfo from '../SectionQuickSearch/SearchInfo/SearchInfo.js';

export default class Card extends Component {

    render() {
        return (
            <div className="card" key={this.props.index}>
              <div className="coverImage" style={{ background: `url('${this.props.image}') center / cover no-repeat` }}>{this.props.title && <span className="textOverlay">{this.props.title}</span>}</div>
              <div className="cardInfo">
                {this.props.place && <HouseInfo place={this.props.place} addToCart={this.props.addToCart}/>}
                {this.props.article && <ArticleInfo article={this.props.article} />}
                {this.props.search && <SearchInfo search={this.props.search} />}
              </div>
            </div>
        );
    }
}

Card.propTypes = {
  index: PropTypes.number,
  image: PropTypes.string.isRequired,
  place: PropTypes.object,
  article: PropTypes.object,
  search: PropTypes.object,
  addToCart: PropTypes.func
}

Card.defaultProps = {
  image: 'test_image',
}
