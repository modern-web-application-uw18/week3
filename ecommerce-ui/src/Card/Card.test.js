import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';

import Card from './Card.js';

describe('Card', () => {
  it('Smoke Test', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Match Snapshot', () => {
    const component = TestRenderer.create(<Card />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  
})
