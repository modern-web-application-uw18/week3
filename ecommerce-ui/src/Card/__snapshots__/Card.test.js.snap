// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Match Snapshot 1`] = `
<div
  className="card"
>
  <div
    className="coverImage"
    style={
      Object {
        "background": "url('test_image') center / cover no-repeat",
      }
    }
  />
  <div
    className="cardInfo"
  />
</div>
`;
